pipelines/pro_image.go

```diff
@@ -0,0 +1,94 @@
+package pipelines
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"dagger.io/dagger"
+	"github.com/grafana/grafana-build/containers"
+)
+
+func ProImage(ctx context.Context, dc *dagger.Client, args PipelineArgs) error {
+	if len(args.PackageInputOpts.Packages) > 1 {
+		return fmt.Errorf("only one package is allowed: packages=%+v", args.PackageInputOpts.Packages)
+	}
+	packages, err := containers.GetPackages(ctx, dc, args.PackageInputOpts, args.GCPOpts)
+	if err != nil {
+		return fmt.Errorf("getting packages: packages=%+v %w", args.PackageInputOpts.Packages, err)
+	}
+
+	debianPackageFile := packages[0]
+
+	log.Printf("Cloning hosted Grafana...")
+	hostedGrafanaRepo, err := containers.CloneWithGitHubToken(dc, args.ProImageOpts.GitHubToken, "https://github.com/grafana/hosted-grafana.git", "main")
+	if err != nil {
+		return fmt.Errorf("cloning hosted-grafana repo: %w", err)
+	}
+
+	socketPath := "/var/run/docker.sock"
+	socket := dc.Host().UnixSocket(socketPath)
+
+	hostedGrafanaImageTag := fmt.Sprintf("hosted-grafana-pro:%s", args.ProImageOpts.ImageTag)
+
+	log.Printf("Building hosted Grafana image...")
+	container := dc.Container().From("golang:1.20-alpine").
```

Comment:
Is there a particular reason why you're not using the docker:cli image here? 🙂

Reply:
No particular reason, should i change it to docker:cli?

Reply:
If the command also works there, I think it would be the better image to use :)

Reply:
Replace `golang:1.20-alpine` with `google/cloud-sdk:433.0.0-alpine` instead of `docker:cli` because we need the gcloud cli. 

What do you think?

Also changed a few other things because the way images were being pushed to the container registry was not correct.